packages/roosterjs-content-model/lib/publicApi/block/toggleMargins.ts

```diff
@@ -0,0 +1,29 @@
+import { getObjectKeys } from 'roosterjs-editor-dom';
+import { createParagraphDecorator } from '../../modelApi/creators/createParagraphDecorator';
+import { MarginFormat } from '../../publicTypes/format/formatParts/MarginFormat';
+import { IContentModelEditor } from '../../publicTypes/IContentModelEditor';
+import { formatParagraphWithContentModel } from '../utils/formatParagraphWithContentModel';
+
+/**
+ * Toggles the current block(s) margin properties.
+ * @param editor The editor to operate on
+ * @param marginFormat Object containing margin props to be changed.
+ * If a margin property is not in this object, it won't be changed.
+ */
+export default function toggleMargins(editor: IContentModelEditor, marginFormat: MarginFormat) {
+    formatParagraphWithContentModel(editor, 'toggleMargins', para => {
+        if (!para.decorator || para.decorator.tagName !== 'p') {
+            para.decorator = createParagraphDecorator('p');
+        }
+        getObjectKeys(marginFormat).forEach(key => {
```

Comment:
what if marginFormat only contain marginTop but existing format contains marginBottom? Do we want to delete it?

getObjectKeys() can only return keys that exists in marginFormat

Reply:
I was going for a function that let us toggle marginTop and/or marginBottom in one implementation. So if a key isn't passed, it will be ignored

Reply:
Let' just have separate parameters and handle them one by one.

Using getObjectKeys() may accidentally overwrite properties that we don't want to if user pass in a larger object then expected parameter type as sub class.